### PR TITLE
Extend ListBox and TreeView samples

### PR DESCRIPTION
**Pull request by KvanTTT**
## What does the pull request do?

ControlCatalog extending by Adding `Add`, `Remove` buttons and `SelectionMode` combobox to ListBoxPage.xaml and to TreeViewPage.xaml.

## What is the current behavior?

The functionality is absence.

## What is the updated/expected behavior with this PR?

Run ContolCatalog and test respective pages.

## Breaking changes

No.

### CODE REVIEW

File: samples/ControlCatalog/Pages/TreeViewPage.xaml.cs [hunk 1]
```diff
@@ -1,40 +1,115 @@
...
         }
 
-        public class Node
+        private class PageViewModel : ReactiveObject
         {
-            private IList<Node> _children;
+            private readonly TreeView _treeView;
+            private SelectionMode _selectionMode;
+
+            public PageViewModel(TreeView treeView)
```
Comment (Gillibald) [inline samples/ControlCatalog/Pages/TreeViewPage.xaml.cs]: I don't like that you pass a control to your viewmodel. Better bind SelectedItems and SelectedItem to a property in your viewmodel and avoid using the TreeView directly that way.

File: samples/ControlCatalog/Pages/ListBoxPage.xaml.cs [hunk 2]
```diff
@@ -1,25 +1,65 @@
...
             AvaloniaXamlLoader.Load(this);
         }
 
+        private class PageViewModel : ReactiveObject
+        {
+            private readonly ListBox _listBox;
+            private int _counter;
+            private SelectionMode _selectionMode;
+
+            public PageViewModel(ListBox listBox)
```
Comment (Gillibald) [inline samples/ControlCatalog/Pages/ListBoxPage.xaml.cs]: Same as with TreeView

**Comment by KvanTTT** [comment]
I think it should be fixed and merged after #2702

**Review by grokys — APPROVED** [review]
There are a few changes I'd like to make here, like 

1. `AlwaysSelected` should be a toggle in addition to the single/multiple etc mode
2. Allow adding items to the middle of the list/tree 

But I'm going to need these things on the `ItemsRepeater` page too, so I will merge this as-is and then update as needed. Thanks @KvanTTT !

**Comment by KvanTTT** [comment]
> AlwaysSelected should be a toggle in addition to the single/multiple etc mode

Ok, maybe it makes sense to add a label `Selection Mode` near this combo box.

> Allow adding items to the middle of the list/tree

It's already supported in TreeView: if at least one item selected, a new item is being inserted as a child of the first selected item.

**Pull request merged**